README.md

```diff
@@ -14,18 +14,70 @@ A proposed [WebAssembly System Interface](https://github.com/WebAssembly/WASI) A
 
 ### Phase 4 Advancement Criteria
 
-`wasi-distributed-lock-service` should have at least two implementations (i.e., from service providers, and or cloud providers), and, at the very minimum, pass the testsuite for Windows, Linux, and MacOS.
+`wasi-distributed-lock` should have at least two implementations (i.e., from service providers, and or cloud providers), and, at the very minimum, pass the test suite for Windows, Linux, and MacOS.
 
 ## Table of Contents
 
 - [Introduction](#introduction)
+- [Goals](#goals)
+- [Non-Goals](#non-goals)
+- [API Walk-through](#api-walk-through)
+  - [Use Case - Leader Election](#use-case---leader-election)
+- [Detailed Design Discussion](#detailed-design-discussion)
+- [Considered Alternatives](#considered-alternatives)
+- [Stakeholder Interest and Feedback](#stakeholder-interest-and-feedback)
+- [References and Acknowledgements](#references-and-acknowledgements)
 
 ### Introduction
 
-The `wasi-distributed-lock-service` world aim to provide a set of generic interfaces for distributed lock services which provide mechanisms to ensure that only one process can mutate a particular state at a time. They are often used to prevent race conditions in scenarios where two or more processes attempting to update the same state at the same time that could result in data inconsistencies.
+The `wasi-distributed-lock` world aim to provide a set of generic interfaces for distributed locking which provides mechanisms to ensure that only one process can mutate a particular state at a time. They are often used to prevent race conditions in scenarios where two or more processes attempting to update the same state at the same time that could result in data inconsistencies.
```

Comment:
```suggestion
The `wasi-distributed-lock` world aims to provide a set of generic interfaces for distributed locking which provides mechanisms to ensure that only one process can mutate a particular state at a time. They are often used to prevent race conditions in scenarios where two or more processes attempting to update the same state at the same time that could result in data inconsistencies.
```

~nit